app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills: list[str] = []
+
+    def learn_skill(self, skill: str) -> None:
+        if skill not in self.skills:
+            self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills.extend(["JavaScript", "HTML", "CSS"])
```

Comment:
The **self.skills** attribute is already initialized as an empty list in the parent **SoftwareEngineer** class when the **__init__** method is called. In child classes, such as FrontendDeveloper, there is no need to create it again.
The use of `self.skills.extend(...`) in child classes is correct because it simply extends the existing list of skills inherited from the parent class.